src/expr/node.h

```diff
@@ -1331,6 +1360,117 @@ TypeNode NodeTemplate<ref_count>::getType(bool check) const
   return NodeManager::currentNM()->getType(*this, check);
 }
 
+template <bool ref_count>
+inline Node NodeTemplate<ref_count>::substituteCaptureAvoiding(
+    Node node, Node replacement) const
+{
+  if (node == *this)
+  {
+    return replacement;
+  }
```

Comment:
I guess one other quick check that we may want to do is `node == replacement ---> return *this`.